src/liboslexec/batched_llvm_gen.cpp

```diff
@@ -6473,6 +6496,217 @@ LLVMGEN (llvm_gen_spline)
+    func_spec.mask();
+
+    // make sure that any temps created to widen uniform values 
+    // are not released until we are done using them!
+    BatchedBackendLLVM::TempScope temp_scope(rop);    
+
+    llvm::Value *args[6] = {
+                            sg_ptr,
+                            rop.llvm_void_ptr (Result),
+                            id_int, 
+                            size_int, 
+                            nullptr,
+                            nullptr
+    };
+    if (weighted) {
+        if (weighted && weight_is_uniform) {
+            args[4] =  rop.llvm_widen_value_into_temp(*weight,0);
+        } else {
+            args[4] = rop.llvm_void_ptr(*weight);
+        }
+
+        args[5] = rop.ll.mask_as_int(rop.ll.current_mask());
+        rop.ll.call_function (rop.build_name(func_spec), cspan<llvm::Value *>(args, 6));
+    } else {
+        args[4] = rop.ll.mask_as_int(rop.ll.current_mask());
+        rop.ll.call_function (rop.build_name(func_spec), cspan<llvm::Value *>(args, 5));
+    }
+
+    llvm::Value *comp_void_ptr = rop.llvm_get_pointer(Result);
+
+    llvm::Value *comp_wide_ptr_ptr = rop.ll.ptr_cast(comp_void_ptr, rop.ll.type_ptr (rop.llvm_type_closure_component_wide_ptr()));
+    llvm::Value *comp_wide_ptr = rop.ll.op_load(comp_wide_ptr_ptr);
+
+    // This is an unrolled vector-width loop.  It was unrolled because it's
+    // was more expedient to write the for in C++ than IR, but if compiliation
+    // of the extra ops becomes problematic, this choice can be revisited
+    for (int lane_index = 0; lane_index < rop.vector_width(); ++lane_index)
+    {
+        llvm::Value *comp_ptr = rop.ll.op_extract(comp_wide_ptr, lane_index);
+        // Get the address of the primitive buffer, which is the 2nd field
+        llvm::Value *mem_void_ptr = rop.ll.GEP (comp_ptr, 0, 2);;
+        mem_void_ptr = rop.ll.ptr_cast(mem_void_ptr, rop.ll.type_void_ptr());
+
+        // For the weighted closures, we need a surrounding "if" so that it's safe
+        // for osl_allocate_weighted_closure_component to return NULL (unless we
+        // know for sure that it's constant weighted and that the weight is
+        // not zero).
+        llvm::BasicBlock *next_block = NULL;
+        llvm::BasicBlock *notnull_block = rop.ll.new_basic_block ("non_null_closure");
+        next_block = rop.ll.new_basic_block ("");
+        llvm::Value *cond = rop.ll.op_ne (rop.ll.ptr_cast(comp_ptr,rop.ll.type_void_ptr()), rop.ll.void_ptr_null());
+        rop.ll.op_branch (cond, notnull_block, next_block);
+        
+        // If the closure has a "prepare" method, call
+        // prepare(renderer, id, memptr).  If there is no prepare method, just
+        // zero out the closure parameter memory.
+        if (clentry->prepare) {
+            // Call clentry->prepare(renderservices *, int id, void *mem)
+            llvm::Value *funct_ptr = rop.ll.constant_ptr((void *)clentry->prepare, rop.llvm_type_prepare_closure_func());
```

Comment:
Was(is) there a unit test that exercised the prepare function?

Reply:
AFAIK There is not -- there isn't anything that appears to exercise it in the single-point testsuite either.  I did a best-effort at making it match the single point implementation.   Maybe Larry or someone else can comment on if this is a critical feature or not?